### DISCUSSION TITLE
Detect window placement relative to monitor

### DISCUSSION THREAD

**Discussion by nickvonkaenel** [shortcuts/no-neck-pain.nvim#412] [Ideas]
Loving this plug-in, thank you so much for your work on it. I generally have my terminal centered on the screen, but occasionally I will add another window to the screen which means it has to be on the right or left (although maybe still taking up 2/3 of the space). I was wondering if there is a way to attempt to center the buffer window closest to the center of the actual monitor. Like if the terminal is more to the left then no-neck-pain only adds padding to the left side and vice-versa.

**Comment by shortcuts** [comment]
Hey, thanks for using the plugin :)

I don't think this is possible, or at least not programatically to fit everyone's use cases, however it might be doable by combining nvim api and no-neck-pain api:

With nvim api, register a custom autocmd for `VimResized` events (assuming moving your window to the side of your screen triggers it), then use nnp api to close the side closer to the center of your screen (e.g. if you move your nvim on the left, you can close the right padding buffer with `NoNeckPainToggleRightSide`), then apply a resize that matches your desired width with `NoNeckPainResize`

I'm not sure how doable it is but that's the "naive" approach I'd go for!

**Reply by nickvonkaenel** [reply]
I believe you are correct, this is trickier than I thought, since there doesn't appear to be a simple way to get the position of an nvim window in screen space without some external api (at least that I could find)

Thanks for the response! I can probably whip up a shortcut with the nnp api to manually reposition things as needed.
